tests/e2e/tests/bookinfo/demo_test.go

```diff
@@ -37,22 +37,22 @@ import (
 const (
 	u1                                 = "normal-user"
 	u2                                 = "test-user"
-	bookinfoYaml                       = "samples/bookinfo/kube/bookinfo.yaml"
-	bookinfoRatingsv2Yaml              = "samples/bookinfo/kube/bookinfo-ratings-v2.yaml"
-	bookinfoRatingsMysqlYaml           = "samples/bookinfo/kube/bookinfo-ratings-v2-mysql.yaml"
-	bookinfoDbYaml                     = "samples/bookinfo/kube/bookinfo-db.yaml"
-	bookinfoMysqlYaml                  = "samples/bookinfo/kube/bookinfo-mysql.yaml"
-	bookinfoDetailsExternalServiceYaml = "samples/bookinfo/kube/bookinfo-details-v2.yaml"
+	bookinfoSampleDir                  = "samples/bookinfo"
+	bookinfoYaml                       = bookinfoSampleDir + "/kube/bookinfo.yaml"
+	bookinfoRatingsv2Yaml              = bookinfoSampleDir + "/kube/bookinfo-ratings-v2.yaml"
+	bookinfoRatingsMysqlYaml           = bookinfoSampleDir + "/kube/bookinfo-ratings-v2-mysql.yaml"
+	bookinfoDbYaml                     = bookinfoSampleDir + "/kube/bookinfo-db.yaml"
+	bookinfoMysqlYaml                  = bookinfoSampleDir + "/kube/bookinfo-mysql.yaml"
+	bookinfoDetailsExternalServiceYaml = bookinfoSampleDir + "/kube/bookinfo-details-v2.yaml"
 	modelDir                           = "tests/apps/bookinfo/output"
-	rulesDir                           = "samples/bookinfo/kube"
-	allRule                            = "route-rule-all-v1.yaml"
-	delayRule                          = "route-rule-ratings-test-delay.yaml"
-	fiftyRule                          = "route-rule-reviews-50-v3.yaml"
-	testRule                           = "route-rule-reviews-test-v2.yaml"
-	testDbRule                         = "route-rule-ratings-db.yaml"
-	testMysqlRule                      = "route-rule-ratings-mysql.yaml"
-	detailsExternalServiceRouteRule    = "route-rule-details-v2.yaml"
-	detailsExternalServiceEgressRule   = "egress-rule-google-apis.yaml"
+	allRule                            = bookinfoSampleDir + "/kube/route-rule-all-v1.yaml"
```

Comment:
I guess this is the real change, for rules

can you put the concatenation once in the code (ie change rulesDir or equiv) rather than add more hardcoded prefixes in all those contants, that seems going in the reverse direction (more hardcoded prefixes rather than less) ?

Reply:
Correct, this is the real change, to enable bookinfo yamls to reside in different directories.

> can you put the concatenation once in the code

Good point, will do.